server/src/server.ts

```diff
@@ -0,0 +1,79 @@
+import express from "express";
+import mongoose from "mongoose";
+import dotenv from "dotenv";
+import cors from "cors";
+
+import { exerciseModel } from "./models/productModel";
+
+dotenv.config();
+const app = express();
+const MONGO_URI = `mongodb+srv://root:${process.env.DBPASS}@gymscheduler.c8rayg3.mongodb.net/`;
+
+app.use(
+  express.json(),
+  cors({
+    origin: process.env.FRONTEND_ADDRESS,
+  })
+);
+
+app.post("/exerciseDay", async (req, res) => {
```

Comment:
Could use some good validation handling here as well: 
https://github.com/jquense/yup
https://www.npmjs.com/package/validator
https://express-validator.github.io/docs/ 
are some popular choices 

It is possible mongoose has its own validators ready too (I didn't use it much), after validation you want to return an error code that lets frontend know what went wrong. 4xx codes mean the client made a mistake they can recover from. 

500 you are throwing here means the server made a mistake and client can do nothing about this. 

Most often in case of validation you throw one of these:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404

Reply:
Added validation for PUT and POST